packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -7790,7 +7790,10 @@ class IndexedSemantics extends SingleChildRenderObjectWidget {
 ///
 /// Useful for attaching a key to an existing widget.
 class KeyedSubtree extends StatelessWidget {
-  /// Creates a widget that builds its child.
+  /// Creates a widget that builds its child while assigning it a key.
+  ///
+  /// This is useful when you want to preserve the state of a widget when it moves
```

Comment:
I'd suggest moving guides like this to the class documentation (L7791), since this explains the entire class, no matter how it's constructed. Constructor docs are more suitable to explain the differences between constructors. For example, for here you might write 
```
See also:

 * [KeyedSubtree.wrap], which is convenient for children that can be indexed and might come with their own keys.
```
(Just an example; you might reword it or ditch it at all.)